ogr/services/github/pull_request.py

```diff
@@ -148,17 +148,18 @@ def create(
         github_repo = project.github_repo
 
         target_project = project
-        if project.is_fork and fork_username is None:
-            logger.warning(f"{project.full_repo_name} is fork, ignoring fork_repo.")
-            source_branch = f"{project.namespace}:{source_branch}"
-            github_repo = project.parent.github_repo
-            target_project = project.parent
-        elif fork_username:
-            source_branch = f"{fork_username}:{source_branch}"
-            if fork_username != project.namespace and project.parent is not None:
-                github_repo = GithubPullRequest.__get_fork(
-                    fork_username, project.parent.github_repo
-                )
+        if project.parent:
+            if project.is_fork and fork_username is None:
```

Comment:
this is exactly one thing that I don't like on static typing :D it has no clue about invariants and integrity

Reply:
You are right, if it is a fork, then it definitely does have a parent but mypy doesn't know anything about it. I am not sure if we can somehow encode this relationship (I don't think so). One more check doesn't hurt and is safer... except that in some cases requre will shout :D